examples/azureml/azureml_submit.py

```diff
@@ -6,54 +6,63 @@
-parser.add_argument("--distributed_config", "-d", type=str, choices=["mpi", "pytorch"], default="pytorch")
-parser.add_argument("--backend", "-b", choices=["sp", "ddp-amp"], default="sp")
-parser.add_argument("--wait", "-w", action="store_true", help="Throw error is Azure ML job fails.")
-args = parser.parse_args()
-
-ws = Workspace.from_config("examples/azureml/config.json")
-
-target = ws.compute_targets[args.target_name]
-
-if args.distributed_config == "pytorch":
-    distributed_job_config = PyTorchConfiguration(
-        process_count=args.process_count, node_count=args.node_count
-    )
-elif args.distributed_config == "mpi":
-    distributed_job_config = MpiConfiguration(
-        process_count_per_node=args.process_count, node_count=args.node_count
+def classification_example():
+    """Prepare the environment and submission command for the classification example."""
+    env = Environment("pymarlin_requirements")
+    env.docker.enabled = True
+    env.docker.base_image = None
+    env.docker.base_dockerfile = 'examples/azureml/dockerfile'
+    env.python.user_managed_dependencies = True
+    env.python.interpreter_path = "/opt/miniconda/bin/python"
+    env.register(ws)
+
+    cmd = f"pip install -U -e . && cd examples/classification/pymarlin_scripts && python train.py --trainer.backend {args.backend}".split()
+    
+    return env, cmd
+    
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--marlin_example", type=str, choices=["classification"], default="classification")
+    parser.add_argument("--target_name", default="sriovdedicated1")
+    parser.add_argument("--node_count", "-n", type=int, default=1)
+    parser.add_argument("--process_count", "-p", type=int, default=1)
+    parser.add_argument("--experiment_name", type=str, default="marlin-tests")
+    parser.add_argument("--distributed_config", "-d", type=str, choices=["mpi", "pytorch"], default="pytorch")
+    parser.add_argument("--backend", "-b", choices=["sp", "ddp-amp"], default="sp")
+    parser.add_argument("--wait", "-w", action="store_true", help="Throw error is Azure ML job fails.")
+    args = parser.parse_args()
+
+    ws = Workspace.from_config("examples/azureml/config.json")
+
+    target = ws.compute_targets[args.target_name]
+
+    if args.distributed_config == "pytorch":
+        distributed_job_config = PyTorchConfiguration(
+            process_count=args.process_count, node_count=args.node_count
+        )
+    elif args.distributed_config == "mpi":
+        distributed_job_config = MpiConfiguration(
+            process_count_per_node=args.process_count, node_count=args.node_count
+        )
+    else:
+        raise ValueError(f"Didn't recognize the distributed config {args.distributed_config}. Select on of 'mpi' or 'pytorch'.")
+
+    if args.marlin_example == "classification":
+        env, cmd = classification_example()
+
+    src = ScriptRunConfig(
+        source_directory='.',
```

Comment:
source_directory should actually still be root of the repo to do a local pip install right? ('../../')

Reply:
@aminsaied could you add a doc under website/docs that goes over how to submit this job with that command? doesn't have to be indepth on azureml but just with that command and that other examples will be added. maybe something small about referencing your AML workspace through config.json and creating your gpu compute (can link to any other resources for AzureML like the cheatsheet or official documentation.)

Reply:
thanks a bunch @aminsaied !